interface/globals.php

```diff
@@ -287,6 +287,14 @@ function undoMagicQuotes($array, $topLevel=true) {
       if ($gl_value == '2') $GLOBALS['sell_non_drug_products'] = 1;
       else if ($gl_value == '3') $GLOBALS['sell_non_drug_products'] = 2;
     }
+    else if ($gl_name == 'gbl_time_zone') {
+      // If value is empty default to old behavior.
+      if ($gl_value) {
+        date_default_timezone_set($gl_value);
+        // Synchronize MySQL time zone with PHP time zone.
+        sqlStatement("SET time_zone = ?", array((new DateTime())->format("P")));
```

Comment:
They don't all go through the sql connection in sql.inc though. Do a quick grep for sql_mode to see the other places where a sql connection is opened up.